sdk/storage/azure_storage_blob/tests/blob_client.rs

```diff
@@ -140,7 +141,7 @@ async fn test_upload_blob(ctx: TestContext) -> Result<(), Box<dyn Error>> {
         .await?;
 
     // Assert
-    let response = blob_client.download_blob(None).await?;
+    let response = blob_client.download(None).await?;
     let content_length = response.content_length()?;
```

Comment:
So this is downloading the entire blob? Not to hold up this PR, and it will likely require changes in `azure_core`, but isn't the idea to get a stream and stream to a file or other caller-provided buffer? Or is that a separate client like ya'll did for Go?

I've been thinking about how to support this case since after beta.2 I want to move deserialization *generally* into the client methods, but we need to make concessions for Storage and maybe Cosmos (/cc @analogrelay). It would be good to chat about that soon.

Again, this isn't to hold up this PR nor the MVP but wanted to call it out / ask about it in context.

Reply:
Yes, this is downloading the entire Blob, unless a range is specificed in the options. And yes, we would ideally want customers to be able to Stream the response body somewhere other than memory. Is there a way to do this with the current `ResponseBody` type in Core? For this test, we chose to just collect the bytes since the Blob was small.

Happy to chat about our requirements whenever.

Reply:
Since this is non-blocking, I will go ahead and resolve in order to merge. Thanks!